### PR TITLE
fix: transaction

### LINKED ISSUE #78: Payout with DOT failing on both prod and test chains

### PR DESCRIPTION
**Pull request by Vovke**
eventually should resolve #78

### CODE REVIEW

**Comment by Vovke** [comment]
resolving #78 and #63

File: src/chain/payout.rs [hunk 1]
```diff
@@ -76,8 +76,33 @@ pub async fn payout(
                 }
             },
             a if (loss_tolerance..=manual_intervention_amount).contains(&a) => {
-                tracing::warn!("Overpayments not handled yet");
-                return Ok(()); //TODO
+                tracing::warn!("Overpayment, proceeding with available balance");
+                // We will transfer all the available balance
```
Comment (Slesarew) [inline src/chain/payout.rs:80]:
```suggestion
                // We will transfer all the available balance
                // TODO smarter handling and returns probably
```

File: src/chain/payout.rs [hunk 2]
```diff
@@ -101,17 +126,23 @@ pub async fn payout(
...
+            }
+        }
 
         let extrinsic = batch_transaction
             .send_this_signed::<(), RuntimeMetadataV15>(&chain.metadata)?
             .ok_or(ChainError::NothingToSend)?;
 
         send_stuff(&client, &format!("0x{}", const_hex::encode(extrinsic))).await?;
 
+        state.order_withdrawn(order.id.clone()).await;
```
Comment (Slesarew) [inline src/chain/payout.rs:145]: is this clone needed?

File: tests/kalatori-api-test-suite/tests/order.test.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ describe('Order Endpoint Blackbox Tests', () => {
     throw new Error('check all environment variables are defined');
   }
   const dotOrderData = {
-    amount: 1,
+    amount: 2,
```
Comment (Slesarew) [inline tests/kalatori-api-test-suite/tests/order.test.ts:11]: add a comment here that indicates that this should be above mortal

**Review by Slesarew — CHANGES_REQUESTED** [review]

File: tests/kalatori-api-test-suite/tests/order.test.ts (hunk 3 above)
Comment (Vovke) [reply tests/kalatori-api-test-suite/tests/order.test.ts]: resolved in [16fe114](https://github.com/Alzymologist/Kalatori-backend/pull/106/commits/16fe114e333990884ec4beabb69f1e5518c74cde)

File: src/chain/payout.rs (hunk 2 above)
Comment (Vovke) [reply src/chain/payout.rs]: resolved in [16fe114](https://github.com/Alzymologist/Kalatori-backend/pull/106/commits/16fe114e333990884ec4beabb69f1e5518c74cde)

File: src/chain/payout.rs (hunk 1 above)
Comment (Vovke) [reply src/chain/payout.rs:80]: resolved in [16fe114](https://github.com/Alzymologist/Kalatori-backend/pull/106/commits/16fe114e333990884ec4beabb69f1e5518c74cde)

**Review by Slesarew — APPROVED** [review]

**Pull request merged**